app/android/app_template/BUILD.gn

```diff
@@ -0,0 +1,67 @@
+import("//build/config/android/rules.gni")
+
+android_apk("xwalk_app_template_apk") {
+  apk_name = "XWalkAppTemplate"
+  android_manifest = "AndroidManifest.xml"
+  java_files = [ "src/org/xwalk/app/template/AppTemplateActivity.java" ]
+  deps = [
+    ":xwalk_app_template_apk_assets",
+    ":xwalk_app_template_apk_resources",
+    "//xwalk/app/android/runtime_client:xwalk_app_runtime_java",
+
+    #TODO(heke123): remove xwalk_core_java, it should be embedded in xwalk_app_runtime_java.
+    "//xwalk/runtime/android/core:xwalk_core_java",
+  ]
+}
+
+android_assets("xwalk_app_template_apk_assets") {
+  sources = []
+  renaming_sources = [
+    "//xwalk/test/android/data/index.html",
+    "//xwalk/test/android/data/sampapp-icon-helloworld.png",
+  ]
+  renaming_destinations = [
+    "www/index.html",
+    "www/sampapp-icon-helloworld.png",
+  ]
+  disable_compression = true
+}
+
+android_resources("xwalk_app_template_apk_resources") {
+  resource_dirs = [ "res" ]
+  custom_package = "org.xwalk.app.template"
+}
+
+action("prepare_xwalk_app_template") {
+  script = "//xwalk/build/android/gn/generate_app_packaging_tool.py"
+  outputs = [
+    "$root_out_dir/xwalk_app_template",
+  ]
+  src_dir = rebase_path("//", "$root_out_dir")
+  args = [
+    "--build-dir=./",
+    "--build-mode=$android_configuration_name",
+    "--output-dir=./xwalk_app_template",
+    "--source-dir=$src_dir/xwalk",
+  ]
+  deps = [
+    ":xwalk_app_template_apk",
+    "//xwalk/app/android/runtime_client:xwalk_app_runtime_java",
+    "//xwalk/build/android:xwalk_core_library",
+    "//xwalk/build/android:xwalk_shared_library",
+  ]
+}
+
+action("xwalk_app_template") {
+  script = "//xwalk/tools/tar.py"
+  outputs = [
+    "$root_out_dir/xwalk_app_template.tar.gz",
+  ]
+  inputs = [
+    "AndroidManifest.xml",
+  ]
+  args = [ "./xwalk_app_template" ]
```

Comment:
I'll remove xwalk_app_template target in this patch because it depends on xwalk_core_library which is still in discussing.